src/memory_manager.rs

```diff
@@ -510,6 +501,20 @@ pub fn initialize_collection<VM: VMBinding>(mmtk: &'static MMTK<VM>, tls: VMThre
     probe!(mmtk, collection_initialized);
 }
 
+/// Ask GC worker threads to exit.  Currently, this function is for the sole purpose of forking.
+///
+/// Arguments:
+/// * `mmtk`: A reference to an MMTk instance.
+pub fn uninitialize_collection<VM: VMBinding>(mmtk: &'static MMTK<VM>) {
```

Comment:
I can't say I like the name `uninitialize_collection` for this API. Can I suggest something like `prepare_to_fork` or just `pre_fork`? If you don't like mentioning `fork` in the API name, then maybe just `stop_gc_threads`?

Reply:
After a second thought, I think this is a good idea.  The `initialize_collection` should be a once-per-MMTk-instance initializer.  Back in JikesRVM, `initialize_collection` was named `enableCollection` (JikesRVM doesn't allow enabling/disabling collection at run time).  It determines the number of threads, and initializes some space-specific data that depends on the number of threads (such as `spillMarkHistograms` in `Defrag`, which we now use `Mutex<Vec<Histogram>>` to avoid making it dependent on the number of GC workers).  It also initialize worker groups, for example, one worker group for parallel workers, and another worker group for concurrent workers, and it is similar to our lxr branch.

Given that `initialize_collection` should be called only once, it is worth making the distinction between `initialize_collection` and the `pre_fork` and `post_fork` functions that may happen as many times as the VM forks.

Reply:
Yes exactly. That's why I felt like `uninitialize_collection` was not reflective of the actual use-case of this API. I pretty much only envision this being used for `fork`, so I think `pre_fork` is appropriate, but if you think it might be used for other reasons, then perhaps just a generic name such as `stop_gc_threads` would be more appropriate.